subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -157,3 +164,63 @@ static int wifi_ap_disable(uint32_t mgmt_request, struct net_if *iface,
 }
 
 NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_AP_DISABLE, wifi_ap_disable);
+
+static int wifi_iface_status(uint32_t mgmt_request, struct net_if *iface,
+			  void *data, size_t len)
+{
+	void *src = NULL;
+	const struct device *dev = net_if_get_device(iface);
+	struct net_wifi_mgmt_offload *off_api =
```

Comment:
Why is the offloading involved here? I mean does the offload API work also with non-offload configurations?

Reply:
Excellent question I just raised another PR to add support for wifi_mgmt offload + native TCP/IP, https://github.com/nrfconnect/sdk-zephyr/pull/880.

Reply:
BTW Isn't the name of that structure confusing? Shouldn't we just rename it to `net_wifi_mgmt? It doesn't really matter where the function is implemented, does it?

Reply:
I also opt for more generic denomination here. It will require changes in other places as well (i.e. non-Nordic drivers) but maybe it's worth to do it right away.

Reply:
If we change then this should be done upstream; surely community will have something to say.